flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -570,7 +595,8 @@ template int Fortran::lower::CharacterOpsBuilder<
 
 template <typename T>
 mlir::Type Fortran::lower::CharacterOpsBuilder<T>::getLengthType() {
-  return mlir::IntegerType::get(64, impl().getContext());
+  // FIXME: make this IndexType? Breaks some tests though.
+  return impl().getIntegerType(64);
```

Comment:
Yes, there are a few helpers that dies because they expect `mlir::IntegerType` and not IndexType.
I am not sure how important it is right now, but it should be possible for it to be one or the other. I have it fixed with len_trim.

Reply:
I think I'm converging on IndexType as the default. As I've simplified FIR over time, it has been easier to just have Ops accept IndexType arguments and dispense with other flavors of integers.

Reply:
Maybe I should expand on this a little.  When no other type has a claim, IndexType seems like a good choice.  In descending order of preference, the alternatives that come to mind are 
- the compilation default integer kind
- kind=8
- kind=4
I believe I prefer IndexType to all of these.  That includes switches that don't really have some other obvious selector type.  Switches that do have a selector type should probably use that type.